src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -172,22 +178,40 @@ function BaseVideoPlayer({
         });
     }, [currentVideoPlayerRef, handleFullscreenUpdate, handlePlaybackStatusUpdate]);
 
+    useLayoutEffect(() => 
+        () => {
+            if(shouldUseSharedVideoElement || videoPlayerRef.current !== currentVideoPlayerRef.current) {
+                return;
+            }
+            currentVideoPlayerRef.current = null;
+        }
+    , [currentVideoPlayerRef, shouldUseSharedVideoElement]);
+
     useEffect(() => {
         if (!isUploading || !videoPlayerRef.current) {
             return;
         }
 
         // If we are uploading a new video, we want to immediately set the video player ref.
+        if(currentVideoPlayerRef.current) {
+            pauseVideo();
+        }
         currentVideoPlayerRef.current = videoPlayerRef.current;
-    }, [url, currentVideoPlayerRef, isUploading]);
+    }, [url, currentVideoPlayerRef, isUploading, pauseVideo]);
 
     // update shared video elements
     useEffect(() => {
-        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL || isFullScreenRef.current) {
+        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL) {
             return;
         }
-        shareVideoPlayerElements(videoPlayerRef.current, videoPlayerElementParentRef.current, videoPlayerElementRef.current, isUploading);
-    }, [currentlyPlayingURL, shouldUseSharedVideoElement, shareVideoPlayerElements, url, isUploading, isFullScreenRef]);
+        shareVideoPlayerElements(videoPlayerRef.current, videoPlayerElementParentRef.current, videoPlayerElementRef.current, isUploading || isPopoverVisible);
```

Comment:
In `shareVideoPlayerElements` function, video is not played if last parameter is true.
So I added `isPopoverVisible` to prevent auto play.